### PR TITLE
add teacher

### PR DESCRIPTION
**Pull request by Bahtah**
Добавил функция создания инструктора

### CODE REVIEW

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/teacher")
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java:17]: api/teachers

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Course.java [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+import kg.peaksoft.peaksoftlmsm1.db.entity.User;
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
 import javax.persistence.*;
 import java.util.Date;
+import java.util.List;
 
 @Data
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/entity/models/Course.java:12]: User @Getter and #Setter instead @Data

File: src/main/java/kg/peaksoft/peaksoftlmsm1/db/service/TeacherService.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    public TeacherResponce delete(Long id){
+        User group = userRepository.findById(id).orElseThrow(() ->
+                new ResourceNotFoundException("Entity", "id", id));
+        userRepository.deleteById(id);
+        return mapToResponse(group);
+    }
+
+    public User mapToEntity(TeacherRequest teacherRequest){
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/db/service/TeacherService.java:52]: Mappers should be in separate class

File: src/main/resources/application-dev.properties [hunk 4]
```diff
@@ -1,7 +1,7 @@
-spring.datasource.url=jdbc:postgresql://localhost:5433/postgres
```
Comment (ChyngyzIshen) [inline src/main/resources/application-dev.properties:1]: No need to add application-dev.properties file in pull request

**Review by ChyngyzIshen — COMMENTED** [review]
There is no @CrossOrigin  and Swagger annotations in TeacherController class

File: src/main/resources/application-dev.properties (hunk 4 above)
Comment (zamiramanasova) [reply src/main/resources/application-dev.properties:1]: @ChyngyzIshen  student has some problem with use dev.properties

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java (hunk 1 above)
Comment (Bahtah) [reply src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java]: Исправил

**Pull request merged**